### PR TITLE
Simplify contract's logics

### PR DESCRIPTION
**Pull request by drinkius**
Updated contract code with some simplifications to avoid possible problems

### CODE REVIEW

File: contracts/Staking.sol [hunk 1]
```diff
@@ -181,48 +150,93 @@ contract Staking is IERC20, IERC20Metadata {
...
+        user.amount += amount;
+        user.rewardDebt = calculateRewardAmount(user.amount);
     }
 
-    function harvest(uint256 _amount) public whenNotPaused {
+    function harvest(uint256 amount) public whenNotPaused {
         updateRewardPool();
         UserInfo storage user = userInfo[msg.sender];
-        // +1 to prevent efforts in scum of tstamp
-        require(user.lastHarvestTimestamp + harvestInterval + 1 > block.timestamp ||
```
Comment (BadConfig) [inline contracts/Staking.sol:205]: The +1 shall be removed before merge

**Review by BadConfig — COMMENTED** [review]
Remove +1 in 173 code line

File: contracts/Staking.sol (hunk 1 above)
Comment (drinkius) [reply contracts/Staking.sol:205]: Good point, removing

**Comment by BadConfig** [comment]
reviewed, sc logic clear, 95+ percent of test coverage is ok

**Review by BadConfig — APPROVED** [review]

**Review by GooGrand — APPROVED** [review]

**Pull request merged**